config/muxcable.py

```diff
@@ -0,0 +1,158 @@
+@click.argument('port', metavar='<port_name>', required=False, default=None)
+def mode(all, status, port):
+    """Show muxcable summary information"""
+
+    state_db = {}
+    appl_db = {}
+    y_cable_tbl = {}
+    port_table_keys = {}
+    y_cable_response_tbl = {}
+
+    # Getting all front asic namespace and correspding config and state DB connector
+
+    namespaces = multi_asic.get_front_end_namespaces()
+    for namespace in namespaces:
+        asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+        state_db[asic_id] = swsscommon.DBConnector("STATE_DB", REDIS_TIMEOUT_MSECS, True, namespace)
+        appl_db[asic_id] = swsscommon.DBConnector("APPL_DB", REDIS_TIMEOUT_MSECS, True, namespace)
+        y_cable_tbl[asic_id] = swsscommon.Table(state_db[asic_id], "MUX_CABLE_TABLE_NAME")
+        port_table_keys[asic_id] = y_cable_tbl[asic_id].getKeys()
+        y_cable_response_tbl[asic_id] = swsscommon.Table(appl_db[asic_id], "MUX_CABLE_RESPONSE_TABLE")
+
+    if port is not None:
+        asic_index = platform_sfputil.get_asic_id_for_logical_port(port)
+        click.echo("asic_index {} \n".format(asic_index))
+        if asic_index is None:
+            click.echo("Got invalid asic index for port {}, cant retreive mux status".format(port))
+            return
+
+        y_cable_asic_table = y_cable_tbl.get(asic_index, None)
+        if y_cable_asic_table is not None:
+            y_cable_asic_table_keys = y_cable_asic_table.getKeys()
+            if port in y_cable_asic_table_keys:
+                port_status_dict = {}
+                (status, fvs) = y_cable_asic_table.get(str(port))
+                if status is not True:
+                    click.echo("could not retrieve port values for port".format(port))
+                fvp = dict(fvs)
+                status_value = fvp.get("status", None)
+                if status == "active" and status_value == "active":
+                    # status is already active, so right back ok
+                    fvs = swsscommon.FieldValuePairs([('status', 'Ok')])
+                    y_cable_response_tbl[asic_index].set(logical_port_name, fvs)
+                    port_status_dict = {'status': 'Ok'}
+                elif status == "active" and status_value == "standby":
+                    # Change of status recived, right back inprogress
+                    fvs = swsscommon.FieldValuePairs([('status', 'inprogress')])
+                    y_cable_response_tbl[asic_index].set(logical_port_name, fvs)
+                    port_status_dict = {'status': 'inprogress'}
+                else:
+                    # Everything else to be treated as failure
+                    fvs = swsscommon.FieldValuePairs([('status', 'Failed')])
+                    y_cable_response_tbl[asic_index].set(logical_port_name, fvs)
+                    port_status_dict = {'status': 'Failed'}
+
+                click.echo("muxcable Ports status : \n {}".format(json.dumps(port_status_dict, indent=4)))
```

Comment:
I think we should consider allowing a `--json` flag to output results in JSON format, otherwise output in a human-readable tabular format using `tabulate`.

Reply:
fixed